skimage/measure/segmentation_metrics.py

```diff
@@ -0,0 +1,493 @@
+            'compare_split_vi',
+            'compare_vi',
+          ]
+
+def compare_raw_edit_distance(im_true, im_test, size_threshold=1000):
+    """Compute the edit distance between two segmentations.
+
+    Parameters
+    ----------
+    im_true : ndarray
+        Ground-truth image.
+    im_test : ndarray
+        Test image.
+    size_threshold : int or float, optional
+        Ignore splits or merges smaller than this number of voxels.
+
+    Returns
+    -------
+    (false_merges, false_splits) : float
+        The number of splits and merges required to convert aseg to gt.
+    """
+    _assert_compatible(im_true, im_test)
+
+    im_test = relabel_from_one(im_test)[0]
+    im_true = relabel_from_one(im_true)[0]
+    r = contingency_table(im_true, im_test)
+    r.data[r.data <= size_threshold] = 0
+    # make each segment overlap count for 1, since it will be one
+    # operation to fix (split or merge)
+    r.data[r.data.nonzero()] /= r.data[r.data.nonzero()]
+    false_splits = (r.sum(axis=0)-1)[1:].sum()
+    false_merges = (r.sum(axis=1)-1)[1:].sum()
+    return (false_merges, false_splits)
+
+def compare_adapted_rand_error(im_true, im_test):
+    """Compute Adapted Rand error as defined by the SNEMI3D contest [1]
+
+    Formula is given as 1 - the maximal F-score of the Rand index
+    (excluding the zero component of the original labels). Adapted
+    from the SNEMI3D MATLAB script, hence the strange style.
+
+    Parameters
+    ----------
+    im_true : ndarray
+        Ground-truth image.
+    im_test : ndarray
+        Test image.
+
+    Returns
+    -------
+    are : float
+        The adapted Rand error; equal to $1 - \frac{2pr}{p + r}$,
+        where $p$ and $r$ are the precision and recall described below.
+    prec : float
+        The adapted Rand precision.
+    rec : float
+        The adapted Rand recall.
+
+    References
+    ----------
+    [1]: http://brainiac2.mit.edu/SNEMI3D/evaluation
+    """
+    _assert_compatible(im_true, im_test)
+
+    # segA is query, segB is truth
+    segA = im_test
+    segB = im_true
+
+    n = segA.size
+
+    # This is the contingency table obtained from segA and segB, we obtain
+    # the marginal probabilities from the table.
+    p_ij = contingency_table(segB, segA)
+
+    # Sum of the joint distribution squared
+    sum_p_ij = p_ij.data @ p_ij.data
+
+    # These are the axix-wise sums (np.sumaxis)
+    a_i = p_ij.sum(axis=0).A.ravel()
+    b_i = p_ij.sum(axis=1).A.ravel()
+
+    # Sum of the segment labeled 'A'
+    sum_a = a_i @ a_i
+    # Sum of the segment labeled 'B'
+    sum_b = b_i @ b_i
+
+    # This is the new code, wherein 'n' is subtacted from the numerator
+    # and the denominator.
+
+    precision = (sum_p_ij - n)/ (sum_a - n)
+    recall = (sum_p_ij - n)/ (sum_b - n)
+
+    fscore = 2. * precision * recall / (precision + recall)
+    are = 1. - fscore
+
+    return (are, precision, recall)
+
+def compare_rand_by_threshold(im_true, im_test, npoints=None):
```

Comment:
We probably want to adapt these names for scikit-image more specifically:

```
im_true -> image_reference
im_test -> image_test
npoints -> n_points
```

But actually, remove `rand_by_threshold` altogether. :joy:

Reply:
I like `pytorch` convention: `input_` and `target`.

Reply:
Another option is: `image_pred`, `image_true`.

Reply:
https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/simple_metrics.py#L46

The convention in this file was `im_true, im_test`, is this different elsewhere in skimage code?